path_collector.go

```diff
@@ -115,6 +110,20 @@ func (c *FilepathCollector) walkDirectoryForYaml(dir string) ([]string, error) {
 	return paths, err
 }
 
+func (c *FilepathCollector) extensionMatches(name string) bool {
+	for _, ext := range c.Extensions {
+		// Users may specify "yaml", but we only want to match ".yaml", not "buyaml".
+		if !strings.HasPrefix(ext, ".") {
+			ext = "." + ext
+		}
+
+		if strings.HasSuffix(name, ext) {
+			return true
+		}
```

Comment:
What about using https://pkg.go.dev/path/filepath#Ext ?

Reply:
This PR is meant to add functionality to support multiple sub-section extensions (i.e. `.yaml.gotmpl`). This function will only return the [extension after the final dot](https://go.dev/play/p/BCG-tbET6Nq).

Reply:
> Ext returns the file name extension used by path. The extension is the suffix beginning at the final dot in the final element of path

`filepath.Ext` won't work for my needs: for `config.yaml.gotmpl` it will return `gotmpl`, not `yaml.gotmpl` which I'd want to match against.